pkg/pxc/app/pmm.go

```diff
@@ -52,9 +53,125 @@ func PMMClient(spec *api.PMMSpec, secrets string, v120OrGreater bool) corev1.Con
 		container.Ports = ports
 	}
 
+	if v170OrGreater {
+		container.LivenessProbe = &corev1.Probe{
+			InitialDelaySeconds: 60,
+			TimeoutSeconds:      5,
+			PeriodSeconds:       10,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port: intstr.FromInt(7777),
+					Path: "/local/Status",
+				},
+			},
+		}
+		container.Env = append(container.Env, pmmAgentEnvs(spec.ServerHost, spec.ServerUser, secrets)...)
+		container.Lifecycle = &corev1.Lifecycle{
+			PreStop: &corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: []string{"bash", "-c", "pmm-admin inventory remove node --force $(pmm-admin status --json | python -c \"import sys, json; print(json.load(sys.stdin)['pmm_agent_status']['node_id'])\")"},
+				},
+			},
+		}
+	}
+
 	return container
 }
 
+func pmmAgentEnvs(pmmServerHost, pmmServerUser, secrets string) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPASE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_ADDRESS",
+			Value: pmmServerHost,
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_USERNAME",
+			Value: pmmServerUser,
+		},
+		{
+			Name: "PMM_AGENT_SERVER_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: SecretKeySelector(secrets, "pmmserver"),
+			},
+		},
+		{
+			Name:  "PMM_AGENT_LISTEN_PORT",
+			Value: "7777",
+		},
+		{
+			Name:  "PMM_AGENT_PORTS_MIN",
+			Value: "30100",
+		},
+		{
+			Name:  "PMM_AGENT_PORTS_MAX",
+			Value: "30105",
+		},
+		{
+			Name:  "PMM_AGENT_CONFIG_FILE",
+			Value: "/usr/local/percona/pmm2/config/pmm-agent.yaml",
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_INSECURE_TLS",
+			Value: "1",
```

Comment:
You don't expect anyone to use valid TLS certificate?

Reply:
for this moment, yes. In the future we will add possibility to configure pmm client more flexible

Reply:
@AlekSi is it configurable via `pmm-admin add` flags?

Reply:
@AlekSi if the user can overwrite it via `pmm-admin add .. --tls` we have added the possibility to specify custom  flags for `pmm-admin add`

Reply:
`pmm-admin add` has nothing to do with that. Than environment variable affects pmm-agent<->pmm-managed connection

Reply:
@spron-in should we have the possibility to customise/manage this option via CR in this release or not?

Reply:
Lets skip in this release.

Reply:
ok